### PR TITLE
ALPINE.txt,Dockerfile{.base},build_docker.sh: bump alpine

### PR DESCRIPTION
**Pull request by irbekrm**
Bump alpine base image version used to build tailscale/tailscale and tailscale/k8s-operator images `3.16` -> `3.18`

Merging this PR will result in our build infra using `alpine:3.18` tag for building both the base image and the containerboot and operator, so the next steps should be 1) build a new base 2) build new containerboot and operator.

I have tested this PR by building containerboot and operator and testing it with the operator ingress/egress proxies.

The release notes for `alpine:3.17` and `alpine:3.18` don't suggest anything potentially breaking:
https://www.alpinelinux.org/posts/Alpine-3.17.0-released.html
https://www.alpinelinux.org/posts/Alpine-3.18.0-released.html

The current latest alpine base is `3.19`, however, we first need to figure out what to do with their swapping `iptables-legacy` -> `iptables-nft` in `3.19`, which seem like would make our images unusable for folks on GKE with COS images and any others that run on hosts that don't support nftables, see https://github.com/tailscale/tailscale/issues/10540#issuecomment-1849034576)

Updates #cleanup

### CODE REVIEW

File: Dockerfile.base [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Copyright (c) Tailscale Inc & AUTHORS
 # SPDX-License-Identifier: BSD-3-Clause
 
-FROM alpine:3.16
+FROM alpine:3.18
```
Comment (irbekrm) [inline Dockerfile.base:4]: This is what actually determines what version of alpine will be pulled in

File: ALPINE.txt [hunk 2]
```diff
@@ -1 +1 @@
-3.16
\ No newline at end of file
+3.18
```
Comment (irbekrm) [inline ALPINE.txt:1]: This just determines how our build infra will tag the new base and what base containerboot/operator images should use, it's ran via `./tool/go run ./cmd/printdep -alpine`

File: Dockerfile [hunk 3]
```diff
@@ -66,7 +66,7 @@ RUN GOARCH=$TARGETARCH go install -ldflags="\
       -X tailscale.com/version.gitCommitStamp=$VERSION_GIT_HASH" \
       -v ./cmd/tailscale ./cmd/tailscaled ./cmd/containerboot
 
-FROM alpine:3.16
+FROM alpine:3.18
```
Comment (irbekrm) [inline Dockerfile:69]: Not sure if anyone uses this image. Our build infra builds with mkctr via ./build_docker.sh script

Comment (maisem) [reply Dockerfile:69]: I don't think so. We've discussed removing it a few times but never actually did.

**Review by maisem — APPROVED** [review]

**Comment by irbekrm** [comment]
There is now a `tailscale/alpine-base:3.18` and `tailscale/k8-operator:unstable-v1.55.158`, `tailscale/tailscale:unstable-v1.55.158` that have been built with that base.

**Pull request merged**